internal/service/flight.go

```diff
@@ -42,3 +43,33 @@ func (service *Flight) Get(ctx context.Context, id uuid.UUID, available bool) (*
 func (service *Flight) Delete(ctx context.Context, id uuid.UUID) error {
 	return service.repo.Delete(ctx, id)
 }
+
+func (service *Flight) Update(ctx context.Context, req request.Flight, id uuid.UUID) (domain.Flight, error) {
+	flight, err := service.Get(ctx, id, false)
+	if err != nil {
+		return domain.Flight{}, errs.ErrFlightNotFound
+	}
+
+	if !req.StartDate.IsZero() {
+		flight.StartDate = req.StartDate
+	}
+	if !req.EndDate.IsZero() {
+		flight.EndDate = req.EndDate
+	}
+	if req.Departure != "" {
+		flight.Departure = req.Departure
+	}
+	if req.Destination != "" {
+		flight.Destination = req.Destination
+	}
+	if req.Rank != "" {
+		flight.Rank = req.Rank
+	}
+	if req.Price != 0 {
+		flight.Price = req.Price
+	}
+	if req.TotalTickets != 0 {
```

Comment:
You might still should to be able to set price and total tickets to 0. But since we want to check whether it's "real zero" or default value, you should consider using pointers to int. Here's a [bit more about this usage](https://www.reddit.com/r/golang/comments/qdgc1q/pointers_in_structures_for_json/). You also would need to revise `Validate()` method and pay attention regarding dereferencing.